sips/cluster_consensus.md

```diff
@@ -28,143 +41,157 @@ The only validator-dependent field is `CommitteeIndex` and it does not have to b
+// ConsensusRunner is in charge of processing consensus messages and managing the consensus instance. There is one per Cluster
+type ConsensusRunner interface {
+	GetBeaconNode() BeaconNode
+	GetValCheckF() qbft.ProposedValueCheckF
+	GetNetwork() Network
+
+	// StartNewConsensus starts a new consensus instance for the given roles and slot
+	StartNewConsensus(roles []types.BeaconRole, slot spec.Slot) error
+	// ProcessConsensus processes a consensus message
+	ProcessConsensus(msg *qbft.SignedMessage) (decided bool, cd *ConsensusData, error)
+	// HasRunningInstance returns true if there is a running consensus instance
+	HasRunningInstance() bool
+}
 
-<p align="center">
-<img src="./images/cluster_consensus/previous_scheme.drawio.png"  width="50%" height="10%">
-</p>
+type ConsensusRunner struct {
+	Share          *types.Share
+	QBFTController *qbft.Controller
+	BeaconNetwork  types.BeaconNetwork
 
-For the proposed change to take place, different `Validator` objects should use the same `QBFT Instance`. For that, we propose decoupling the `QBFTController` object from the `DutyRunner`.
+	// highestDecidedSlot holds the highest decided duty slot and gets updated after each decided is reached
+	highestDecidedSlot spec.Slot
 
-<p align="center">
-<img src="./images/cluster_consensus/new_scheme.drawio.png"  width="50%" height="10%">
-</p>
+}
 
-### New IDs
+// PartialSigRunner is in charge of aggregating partial signatures and commiting signed data to the beacon node. There is one per Validator.
+type PartialSigRunner interface {
+	GetBeaconNode() BeaconNode
+	GetValCheckF() qbft.ProposedValueCheckF
+	GetSigner() types.BeaconSigner
+	GetNetwork() Network
 
-Since a single `QBFT instance` will be responsible for several validator duties, its ID must not be dependent on a validator key but rather on a cluster of operators. For that, we propose changing the `MessageID` from
+	//TODO should add local duty? or use the one in cd? 
+	UponDecided(cd *ConsesusData) error
+	ProcessPostConsensus(signedMsg *types.SignedPartialSignatureMessage) error
+}
 
-```mermaid
-flowchart RL
-	subgraph MessageID
-		Domain
-		ValidatorPublicKey
-		Role
-	end
+type PartialSigRunner struct
+	State          *State
+	Share          *types.Share
+	BeaconNetwork  types.BeaconNetwork
+	BeaconRoleType types.BeaconRole
 ```
 
-to
-
+#### Happy Flow
 
-```mermaid
-flowchart RL
-	subgraph MessageID
-		Domain
-		OperatorCluster
-		Role
-	end
-```
+1. `Cluster` receives duties that match a certain slot, starts consensus for the relevant roles, and initializes the `PartialSigRunners` for the relevant Validators.
+2. `Cluster` receives consensus messages and hands them over to the `QBFTController` that has unchanged logic.
+3. Once `ProcessConsensus` returns `decided = true`, the `Cluster` will call `UponDecided(cd)` for each `PartialSigRunner` that has been initialized. This will cause an ommision of a partialSigMessage for each validator.
```

Comment:
```suggestion
3. Once `ProcessConsensus` returns `decided = true`, the `Cluster` will call `UponDecided(cd)` for each `PartialSigRunner` that has been initialized. This will cause an omission of a partialSigMessage for each validator.
```